File: Java/src/main/java/Solution701_800/Solution717.java
package Solution701_800;

/**
 * @author windows
 */
public class Solution717 {
    public boolean isOneBitCharacter(int[] bits) {
        if (bits.length == 1){
            return true;
        }
        if (bits[bits.length - 1] != 0){
            return false;
        }

        return false;
    }


    public static void main(String[] args) {
        Solution717 solution717 = new Solution717();
    }
}

File: C++/src/Solution701_800/Solution717.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution717.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool Solution717::isOneBitCharacter(vector<int> &bits) {
    int n = bits.size();
    for (int i = 0; i < n - 1; ++i) {
        if (bits[i] == 1){
            bits[i + 1] = -1;
        }
    }
    return bits[n - 1] == 0;
}
